### PR TITLE
Feat dialog

### PR DESCRIPTION
**Pull request by AdamKong**
dialog feature has finished.

### CODE REVIEW

File: c [hunk 1]
```diff
@@ -0,0 +1,3 @@
+1111
```
Comment (haohaokong) [inline c]: grammar error in this line, please change the code.

**Review by haohaokong — CHANGES_REQUESTED** [review]

File: c (hunk 1 above)
Comment (AdamKong) [reply c]: May I change it to 4444 probably? thanks!

Comment (AdamKong) [reply c]: Ok, now I understand. thanks @haohaokong

Comment (haohaokong) [reply c]: yes, your understanding is correct.

**Pull request merged**